git-odb/src/loose/object/decode.rs

```diff
@@ -94,7 +94,9 @@ impl loose::Object {
             })?;
             self.compressed_data = SmallVec::from(buf);
         }
-        self.decompressed_data = SmallVec::from(decompress_to_vec_zlib(&self.compressed_data[..]).unwrap());
+        let mut vec = Vec::with_capacity(std::cmp::min(self.compressed_data.len() * 2, usize::MAX));
+        Decompress::new(true).decompress_vec(&self.compressed_data[..], &mut vec, flate2::FlushDecompress::None)?;
+        self.decompressed_data = SmallVec::from(vec);
```

Comment:
This is what `decompress_to_vec_zlib` does behind the scenes, if my understanding of their source code is correct.

On that note--do we actually use `SmallVec` properly here? Originally and on L95 we're instantiating an instance of `SmallVec` from a `Vec`, which means that `Vec` already performs the heap allocation if it's non-empty, and so we never actually inline anything?

Reply:
I agree, there it would by necessity reuse the given `Vec` and if anything add a layer that does nothing for us.

If there is no other, correct, use of that smallvec I think it would better be removed. Please note that I didn't look into it yet but will try soon.